tests/setuptab.test.js

```diff
@@ -260,116 +264,94 @@ describe('UI spec functionality', () => {
         arg4: {
           name: 'Dfoo',
           type: 'number',
-        },
-        arg5: {
-          name: 'Efoo',
-          type: 'number',
-        },
-      },
+        }
+      }
     };
 
     const uiSpec = {
-      controller: {
-        ui_control: ['arg2', 'arg3', 'arg4'],
-      },
-      arg2: {
-        ui_option: 'disable',
-      },
-      arg3: {
-        ui_option: 'hide',
-      },
-      arg4: {
-        ui_option: 'foo', // an invalid option should be ignored
-      },
+      order: [['arg1', 'arg2'], ['arg3', 'arg4']],
+      enabledFunctions: {
+        // enabled if arg1 is sufficient
+        arg2: (state => state.argsEnabled['arg1'] && !!state.argsValues['arg1'].value),

```

Comment:
I notice you're not using the `isSufficient` and `isNotSufficient` helper functions here, maybe because they're not in scope. I've run into this problem as well and I've sometimes decided to export additional objects from a module just so I could re-use them in a test, or so that I could test them in a unit test. So if you thought it was beneficial, you could consider exporting those functions from `ui_config.js` and importing them in this test. I don't love that - exporting functions that aren't used anywhere outside the module where they're defined - but it can be convenient for tests.

Reply:
I feel like the helper functions are simple enough that it's okay to replicate them here rather than exporting. I'm not really trying to test the helper functions themselves here. Maybe I should, but they're almost too basic to be worth it.